From e4fc870f38c04be2f02580fadcca64d3c1dd7c61 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Fri, 9 Mar 2018 07:13:55 +0100
Subject: [PATCH 13/51] jni/libfm_jni: Resolve FM_DEVICE_PATH R/O open flag

 * libfm_jni/FmRadioController.cpp:96:58:
    error: 'open' has superfluous mode bits; missing O_CREAT?

 * jni/android_hardware_fm.cpp:668:47:
    error: 'open' has superfluous mode bits; missing O_CREAT?

Change-Id: I3f4f3bf4e27ea55539ec81b2e59b17be6dcb0892
---
 jni/android_hardware_fm.cpp     | 2 +-
 libfm_jni/FmRadioController.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/jni/android_hardware_fm.cpp b/jni/android_hardware_fm.cpp
index d63d416..86624b2 100644
--- a/jni/android_hardware_fm.cpp
+++ b/jni/android_hardware_fm.cpp
@@ -659,7 +659,7 @@ static jint android_hardware_fmradio_FmReceiverJNI_acquireFdNative
     if(radio_path == NULL){
         return FM_JNI_FAILURE;
     }
-    fd = open(radio_path, O_RDONLY, O_NONBLOCK);
+    fd = open(radio_path, O_RDONLY, 0);
     if(isCopy == JNI_TRUE){
         env->ReleaseStringUTFChars(path, radio_path);
     }
diff --git a/libfm_jni/FmRadioController.cpp b/libfm_jni/FmRadioController.cpp
index a002121..ffcfb32 100644
--- a/libfm_jni/FmRadioController.cpp
+++ b/libfm_jni/FmRadioController.cpp
@@ -93,7 +93,7 @@ int FmRadioController ::open_dev()
 {
     int ret = FM_SUCCESS;
 
-    fd_driver = open(FM_DEVICE_PATH, O_RDONLY, O_NONBLOCK);
+    fd_driver = open(FM_DEVICE_PATH, O_RDONLY, 0);
 
     if (fd_driver < 0) {
         ALOGE("%s failed, [fd=%d] %s\n", __func__, fd_driver, FM_DEVICE_PATH);
-- 
2.17.1

